connectors/src/connectors/snowflake/temporal/client.ts

```diff
@@ -0,0 +1,88 @@
+import type { ModelId, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { WorkflowHandle } from "@temporalio/client";
+import { WorkflowNotFoundError } from "@temporalio/client";
+
+import { QUEUE_NAME } from "@connectors/connectors/snowflake/temporal/config";
+import { snowflakeSyncWorkflow } from "@connectors/connectors/snowflake/temporal/workflows";
+import { getTemporalClient } from "@connectors/lib/temporal";
+import logger from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+function makeSnowflakeSyncWorkflowId(connectorId: ModelId): string {
+  return `snowflake-sync-${connectorId}`;
+}
+
+export async function launchSnowflakeSyncWorkflow(
+  connectorId: ModelId
+): Promise<Result<string, Error>> {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    throw new Error(
+      `[Snowflake] Connector not found. ConnectorId: ${connectorId}`
+    );
+  }
+
+  const client = await getTemporalClient();
+  const workflowId = makeSnowflakeSyncWorkflowId(connectorId);
+
+  try {
+    await client.workflow.start(snowflakeSyncWorkflow, {
```

Comment:
Could we use signals so that it's easy to upsert permissions & to handle full resync? 
(i.e using workflow.signalWithStart)

Reply:
I'm not fully sure I understand what it would do -- immediately launch without waiting for cron to fire ?